src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -228,28 +254,63 @@ public Path exportProgrammingExerciseRepositories(ProgrammingExercise exercise,
      * @return a zipped file
      */
     public Optional<File> exportInstructorRepositoryForExercise(long exerciseId, RepositoryType repositoryType, Path outputDir, List<String> exportErrors) {
+        var exerciseOrEmpty = loadExerciseForRepoExport(exerciseId, repositoryType.getName(), exportErrors);
+        if (exerciseOrEmpty.isEmpty()) {
+            return Optional.empty();
+        }
+        var exercise = exerciseOrEmpty.get();
+        String zippedRepoName = getZippedRepoName(exercise, repositoryType.getName());
+        var repositoryUrl = exercise.getRepositoryURL(repositoryType);
+        return exportRepository(repositoryUrl, repositoryType.getName(), zippedRepoName, exercise, outputDir, exportErrors);
+    }
+
+    /**
+     * Exports a repository available for an instructor/tutor for a given programming exercise. This can be a template,
+     * solution, or tests repository
```

Comment:
This cannot be a template, solution or test repository, right?

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/3457/commits/415d2021c642b223e9aa038d860cd48e3f706801